EssentialsProtect/src/com/earth2me/essentials/protect/EssentialsProtectBlockListener.java

```diff
@@ -90,4 +97,20 @@ public void onPortalLight(PortalCreateEvent event) {
             event.setCancelled(prot.getSettingBool(ProtectConfig.prevent_portal_creation));
         }
     }
+
+    @EventHandler(priority = EventPriority.HIGHEST, ignoreCancelled = true)
+    public void onSignDye(PlayerInteractEvent event) {
+
+        if (VersionUtil.getServerBukkitVersion().isLowerThan(VersionUtil.v1_14_R01)) return;
+
+        Material clickedBlock = event.getClickedBlock().getBlockData().getMaterial();
+        Action action = event.getAction();
+        User user = ess.getUser(event.getPlayer());

```

Comment:
Why use `User`? `PlayerInteractEvent#getItem` returns the stack that the player used.